### PR TITLE
feat: Add `/cardano/sync_state` endpoint, refactor

### PR DESCRIPTION
**Pull request by Mr-Leshiy**
# Description

- added new `/cardano/sync_state` endpoint which return current follower's sync state.
- refactored cardano related endpoints and types
- updated `cardano-chain-follower` to the patched version (later will be replaced to the main, after it will be completely merged)

related to #309

### CODE REVIEW

File: catalyst-gateway/bin/src/event_db/config.rs [hunk 1]
```diff
@@ -1,20 +1,23 @@
...
 use serde::{Deserialize, Serialize};
 
 use crate::event_db::{Error, EventDB};
 
-#[derive(Serialize, Deserialize, Debug, PartialEq, PartialOrd, Clone)]
+#[derive(Debug, PartialEq, Clone)]
 /// Network config metadata
 pub(crate) struct FollowerConfig {
     /// Mainnet, preview, preprod
-    pub(crate) network: String,
```
Comment (minikin) [inline catalyst-gateway/bin/src/event_db/config.rs:10]: Can you please elaborate on why you switched from `Network` type to `String`?

File: catalyst-gateway/bin/src/event_db/config.rs [hunk 2]
```diff
@@ -40,7 +43,8 @@ impl EventDB {
 
         let mut follower_configs = Vec::new();
         for row in rows {
-            let network = row.try_get("id3")?;
```
Comment (minikin) [inline catalyst-gateway/bin/src/event_db/config.rs:43]: Can we refactor the magic string `id3` to something more concrete and self-explanatory?

**Review by minikin — CHANGES_REQUESTED** [review]

File: catalyst-gateway/bin/src/service/api/cardano/sync_state_get.rs [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+pub(crate) type AllResponses = response! {
+    200: OK<Json<Option<SyncState>>>,
+    404: NotFound,
+    500: ServerError,
+    503: ServiceUnavailable,
+};
+
+/// # GET `/utxo/staked_ada`
+#[allow(clippy::unused_async)]
+pub(crate) async fn endpoint(state: &State, network: Option<Network>) -> AllResponses {
```
Comment (minikin) [inline catalyst-gateway/bin/src/service/api/cardano/sync_state_get.rs:33]: We can improve this function by early returns for error handling, reducing nesting, and improving the reliability of the function. For example:

```rust
#[allow(clippy::unused_async)]
pub(crate) async fn endpoint(state: &State, network: Option<Network>) -> AllResponses {
    // Immediately convert network to its non-optional form to simplify further logic.
    let network = network.unwrap_or(Network::Mainnet);
    
      // Define a closure for handling repetitive error formatting.
    let server_error_response = |err| T500(server_error!("{}", err.to_string()));

    let event_db = match state.event_db() {
        Ok(db) => db,
        Err(Error::EventDb(DBError::MismatchedSchema { was, expected })) => {
            tracing::error!(
                expected = expected,
                current = was,
                "DB schema version status mismatch"
            );
            state.set_schema_version_status(SchemaVersionStatus::Mismatch);
            return T503(ServiceUnavailable);
        },
        Err(err) => return server_error_response(err),
    };

    // Async block to allow using `await` with early returns. I'm not 100% if this is required.
    async move {
        match event_db.last_updated_metadata(network.into()).await {
            Ok((slot_number, block_hash, last_updated)) => T200(OK(Json(Some(SyncState {
                slot_number,
                block_hash,
                last_updated,
            })))),
            Err(DBError::NotFound) => T404(NotFound),
            Err(err) => server_error_response(err),
        }
    }
    .await
}
```

File: catalyst-gateway/bin/src/event_db/config.rs (hunk 2 above)
Comment (minikin) [reply catalyst-gateway/bin/src/event_db/config.rs:43]: What if we define a new type, like this:

```rust
enum CardanoNetwork {
    Id,
    Id2,
    Id3,
}

impl CardanoNetwork {
    fn as_str(&self) -> &'static str {
        match self {
            CardanoNetwork::Id => "id",
            CardanoNetwork::Id2 => "id2",
            CardanoNetwork::Id3 => "id3",
        }
    }
}


fn main() {
   let cardanoNetwork = CardanoNetwork::Id;
    println!("The CardanoNetwork is: {}", cardanoNetwork.as_str());
}
```

Comment (Mr-Leshiy) [reply catalyst-gateway/bin/src/event_db/config.rs:43]: Yea, thank you, totally make sense

File: catalyst-gateway/bin/src/service/api/cardano/sync_state_get.rs (hunk 3 above)
Comment (Mr-Leshiy) [reply catalyst-gateway/bin/src/service/api/cardano/sync_state_get.rs:33]: done

**Review by minikin — APPROVED** [review]
lgtm

**Pull request merged**